Hello,

I made the two code changes in posterior.py, however, I still get the exact same error.  I'm not sure how to run the "test" using the files within the test folder...

Thanks,
Paul

cellbender:remove-background: Computing target noise counts per gene for MCKP estimator
Traceback (most recent call last):
  File "C:\Users\SuP1\.conda\envs\cellbender\Scripts\cellbender-script.py", line 33, in <module>
    sys.exit(load_entry_point('cellbender', 'console_scripts', 'cellbender')())
  File "c:\users\sup1\downloads\cellbender-0.3.0\cellbender-0.3.0\cellbender\base_cli.py", line 123, in main
    cli_dict[args.tool].run(args)
  File "c:\users\sup1\downloads\cellbender-0.3.0\cellbender-0.3.0\cellbender\remove_background\cli.py", line 185, in run
    return main(args)
  File "c:\users\sup1\downloads\cellbender-0.3.0\cellbender-0.3.0\cellbender\remove_background\cli.py", line 230, in main
    posterior = run_remove_background(args)
  File "c:\users\sup1\downloads\cellbender-0.3.0\cellbender-0.3.0\cellbender\remove_background\run.py", line 133, in run_remove_background
    file_name=file_name,
  File "c:\users\sup1\downloads\cellbender-0.3.0\cellbender-0.3.0\cellbender\remove_background\run.py", line 237, in compute_output_denoised_counts_reports_metrics
    per_gene=True,
  File "c:\users\sup1\downloads\cellbender-0.3.0\cellbender-0.3.0\cellbender\remove_background\posterior.py", line 1579, in compute_mean_target_removal_as_function
    device=device,
  File "c:\users\sup1\downloads\cellbender-0.3.0\cellbender-0.3.0\cellbender\remove_background\estimation.py", line 150, in estimate_noise
    dtype=np.float32)
  File "c:\users\sup1\downloads\cellbender-0.3.0\cellbender-0.3.0\cellbender\remove_background\estimation.py", line 77, in _estimation_array_to_csr
    dtype=dtype,
  File "c:\users\sup1\downloads\cellbender-0.3.0\cellbender-0.3.0\cellbender\remove_background\estimation.py", line 238, in _estimation_array_to_csr
    row, col = index_converter.get_ng_indices(m_inds=m)
  File "c:\users\sup1\downloads\cellbender-0.3.0\cellbender-0.3.0\cellbender\remove_background\posterior.py", line 1537, in get_ng_indices
    raise ValueError(f'Requested m_inds out of range: '
ValueError: Requested m_inds out of range: [-2147483648 -2147483648 -2147483648 ... -2147483648 -2147483648
 -2147483648]

--- a/.github/workflows/run_pytest.yml
+++ b/.github/workflows/run_pytest.yml
@@ -1,4 +1,4 @@
-# Run cellbender's tests
+# Run cellbender test suite
 
 name: 'pytest'
 
@@ -7,10 +7,12 @@ on: pull_request
 jobs:
   build:
 
-    runs-on: 'ubuntu-latest'
     strategy:
       matrix:
+        os: ['ubuntu-latest', 'windows-latest']
         python-version: ['3.7']
+
+    runs-on: ${{ matrix.os }}
 
     steps:
     - name: 'Checkout repo'

--- a/cellbender/remove_background/checkpoint.py
+++ b/cellbender/remove_background/checkpoint.py
@@ -297,7 +297,7 @@ def make_tarball(files: List[str], tarball_name: str) -> bool:
         for file in files:
             # without arcname, unpacking results in unpredictable file locations!
             tar.add(file, arcname=os.path.basename(file))
-    os.rename(tarball_name + '.tmp', tarball_name)
+    os.replace(tarball_name + '.tmp', tarball_name)
     return True
 
 

--- a/cellbender/remove_background/tests/test_posterior.py
+++ b/cellbender/remove_background/tests/test_posterior.py
@@ -396,7 +396,7 @@ def test_save_and_load(tmpdir_factory, blank_noise_offsets, m):
     if blank_noise_offsets:
         noise_offsets = {}
     else:
-        noise_offsets = dict(zip(np.random.randint(low=0, high=(m - 1), size=10),
+        noise_offsets = dict(zip(np.random.randint(low=0, high=(m - 1), size=10, dtype=np.uint64),
                                  np.random.randint(low=1, high=5, size=10)))
     kwargs = {'a': 'b', 'c': 1}
     kwargs2 = {'a': 'method', 'c': 1}

